multiqc/utils/config.py

```diff
@@ -39,12 +41,116 @@
-    configs = yaml.safe_load(f)
-    for c, v in configs.items():
-        globals()[c] = v
+# Declaring variables for static typing
+title: str
+subtitle: str
+intro_text: str
+report_comment: str
+report_header_info: List[Dict[str, str]]
+show_analysis_paths: bool
+show_analysis_time: bool
+config_file: str
+custom_logo: str
+custom_logo_url: str
+custom_logo_title: str
+custom_css_files: List[str]
+simple_output: bool
+template: str
+profile_runtime: bool
+pandoc_template: str
+read_count_multiplier: float
+read_count_prefix: str
+read_count_desc: str
+long_read_count_multiplier: float
+long_read_count_prefix: str
+long_read_count_desc: str
+base_count_multiplier: float
+base_count_prefix: str
+base_count_desc: str
+output_fn_name: str
+data_dir_name: str
+plots_dir_name: str
+data_format: str
+module_tag: List[str]
+force: bool
+no_ansi: bool
+quiet: bool
+prepend_dirs: bool
+prepend_dirs_depth: int
+prepend_dirs_sep: str
+file_list: bool
+require_logs: bool
+
+make_data_dir: bool
+zip_data_dir: bool
+data_dump_file: bool
+megaqc_url: str
+megaqc_access_token: str
+megaqc_timeout: float
+export_plots: bool
+make_report: bool
+plots_force_flat: bool
+plots_force_interactive: bool
+plots_flat_numseries: int
+num_datasets_plot_limit: int
+collapse_tables: bool
+max_table_rows: int
+table_columns_visible: Dict
+table_columns_placement: Dict
+table_columns_name: Dict
+table_cond_formatting_colours: List[Dict[str, str]]
+table_cond_formatting_rules: Dict[str, List[Dict[str, str]]]
+decimalPoint_format: str
+thousandsSep_format: str
+remove_sections: List
+section_comments: Dict
+lint: bool  # Deprecated since v1.17
+strict: bool
+custom_plot_config: Dict
+custom_table_header_config: Dict
+software_versions: Dict
+ignore_symlinks: bool
+ignore_images: bool
+fn_ignore_dirs: List[str]
+fn_ignore_paths: List[str]
+sample_names_ignore: List[str]
+sample_names_ignore_re: List[str]
+sample_names_rename_buttons: List[str]
+sample_names_replace: Dict
+sample_names_replace_regex: bool
+sample_names_replace_exact: bool
+sample_names_replace_complete: bool
+sample_names_rename: List
+show_hide_buttons: List
+show_hide_patterns: List
+show_hide_regex: List
+show_hide_mode: List
+no_version_check: bool
+log_filesize_limit: int
+filesearch_lines_limit: int
+report_readerrors: int
+skip_generalstats: int
+skip_versions_section: int
+disable_version_detection: int
+versions_table_group_header: str
+data_format_extensions: Dict[str, str]
+export_plot_formats: List[str]
+custom_content: Dict
+fn_clean_sample_names: bool
+use_filename_as_sample_name: bool
+fn_clean_exts: List
+fn_clean_trim: List
+fn_ignore_files: List
+top_modules: List
+module_order: List
+
+# Populating the variables above from the default MultiQC config
+config_defaults_path = os.path.join(MULTIQC_DIR, "utils", "config_defaults.yaml")
+with open(config_defaults_path) as f:
+    _default_config = yaml.safe_load(f)
+for c, v in _default_config.items():
+    globals()[c] = v
```

Comment:
![](https://media2.giphy.com/media/3ofT5MUq4vzDjZuVGM/giphy.gif)

Reply:
Haha. That will be refactored with pydantic etc, but for now I really like how just declaring the variables with type hints enabled static typing with a very minimal invasiveness. Nice incremental improvement